conda_build/inspect_pkg.py

```diff
@@ -34,32 +41,35 @@
     rm_rf,
 )
 
+from .deprecations import deprecated
 
+
+@deprecated("3.28.0", "4.0.0")
 @lru_cache(maxsize=None)
-def dist_files(prefix, dist):
+def dist_files(prefix: str | os.PathLike | Path, dist: Dist) -> set[str]:
     meta = is_linked(prefix, dist)
     return set(meta["files"]) if meta else set()
 
 
-def which_package(in_prefix_path, prefix, avoid_canonical_channel_name=False):
+@deprecated.argument("3.28.0", "4.0.0", "avoid_canonical_channel_name")
+def which_package(
+    path: str | os.PathLike | Path,
+    prefix: str | os.PathLike | Path,
+) -> Iterable[PackageRecord]:
     """
-    given the path of a conda installed file iterate over
+    Given the path (of a (presumably) conda installed file) iterate over
     the conda packages the file came from.  Usually the iteration yields
     only one package.
     """
-    norm_ipp = normcase(in_prefix_path.replace(os.sep, "/"))
-    from conda_build.utils import linked_data_no_multichannels
+    from conda.core.prefix_data import PrefixData
 
-    if avoid_canonical_channel_name:
-        fn = linked_data_no_multichannels
-    else:
-        fn = linked_data
```

Comment:
`linked_data_no_multichannels` and `linked_data` do the following:

```python
from conda.core.prefix_data import PrefixData
from conda.models.dist import Dist

def linked_data_no_multichannels(prefix):
    return {
        Dist.from_string(prec.fn, channel_override=prec.channel.name): prec
        for prec in PrefixData(prefix).iter_records()
    }


def linked_data(prefix):
    return {
        Dist(prec): prec
        for prec in PrefixData(prefix).iter_records()
    }
```

since this PR eliminates the poor `PrefixRecord` → `Dist` translation I don't believe there's a need to distinguish between `avoid_canonical_channel_name` anymore (not that I'm entirely sure what that means in the first place)